pkg/controller/secret/secretsync.go

```diff
@@ -161,13 +168,23 @@ func (c *SecretSyncController) translateSecret(cloudSecret *v1.Secret, cloudConf
 	}
 	accountID := match[1]
 
+	iamEndpoint, err := fetchEndpoint("iamEndpointOverride", conf, defaultTokenExchangeURL)
```

Comment:
If you modify `fetchEndpoint` as per my suggestion then error handling is not required